### PR TITLE
added implementation of generating user stats

### PR DESCRIPTION
**Pull request by MichalPleva8**

### CODE REVIEW

**Comment by MichalPleva8** [comment]
Statická url kde sa vždicky nahráva výstupný spvor https://ipfs.synthetix.io/ipns/k2k4r8oeszqrv8tnl0k3bvo8j9f63sci60mp4rms8iy77o9vq96gad29

**Comment by LubomirIgonda1** [comment]
@MichalPleva8 btw the [file ](https://ipfs.synthetix.io/ipns/k2k4r8oeszqrv8tnl0k3bvo8j9f63sci60mp4rms8iy77o9vq96gad29)has >100Kb it would be nice if the names of attributes was in shorter

File: scripts/generateStats/generateUserStats.ts [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+		if (parlayMarketsBatch.data.errors && parlayMarketsBatch.data.errors.length) {
+			throw new Error(`Error: ${parlayMarketsBatch.data.errors.join('; ')}`)
+		}
+
+		const manchesterUnited = [...batch.data.data.parlayMarkets, ...batch.data.data.positionBalances, ...parlayMarketsBatch.data.data.parlayMarkets]
+
+		tickets.push(manchesterUnited)
+	}
+
+	return tickets.flat()
```
Comment (LubomirIgonda1) [inline scripts/generateStats/generateUserStats.ts:125]: @MichalPleva8 why do you need flat data array ?. If there is any reason please write some comment

File: scripts/generateStats/generateUserStats.ts [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+}
+
+const writeStatsToFile = async (stats: Record<string, any>, processStart: string) => {
+	const fileName = `stats-${network.name}-${processStart}.json`
+	const statsFile = await fs.open(path.join(process.cwd(), 'scripts', 'generateStats', 'data', fileName), 'w')
+
+	await statsFile.writeFile(JSON.stringify(stats))
```
Comment (LubomirIgonda1) [inline scripts/generateStats/generateUserStats.ts:132]: @MichalPleva8 this cloud be write with just one line 
`await fs.promises.writeFile(destinationFilePath, JSON.strinigy(data))`

File: scripts/utils/queries.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+				id
+				side
+				market {
+					id
+					address
+					doubleChanceMarketType
+					homeTeam
+					homeScore
+					homeOdds
+					awayOdds
```
Comment (LubomirIgonda1) [inline scripts/utils/queries.ts:70]: @MichalPleva8 do you need all the data ? from queries ?

File: scripts/generateStats/generateUserStats.ts (hunk 1 above)
Comment (MichalPleva8) [reply scripts/generateStats/generateUserStats.ts:125]: @LubomirIgonda1 because each batch (iteration) comes as a array, that need to flattened

**Comment by MichalPleva8** [comment]
@LubomirIgonda1 please check the latest changes 🙂

File: scripts/utils/queries.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+		parlayMarkets(first: $firstParlay, skip: $skipParlay, orderBy: timestamp, orderDirection: $orderDirection) {
+			id
+			claimed
+			sUSDPaid
+			timestamp
+			totalQuote
+			won
+			account
+			totalAmount
+			positions(first: 10, skip: 0) {
```
Comment (LubomirIgonda1) [inline scripts/utils/queries.ts:12]: @MichalPleva8  does not need to be paginated  ?

**Review by LubomirIgonda1 — APPROVED** [review]

**Pull request merged**